lib/simp/beaker_helpers.rb

```diff
@@ -708,7 +710,7 @@ def install_puppet
 
     unless puppet_agent_version
       if host.options['puppet_collection'] && host.options['puppet_collection'] =~ /puppet(\d+)/
-        puppet_agent_version = latest_puppet_agent_version_for($1)
+        puppet_agent_version = latest_puppet_agent_version_for("~> #{$1}")
```

Comment:
What if the user really just wants a specific version, e.g., 4.10.2?  This would bring in 4.10.7.